krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -17,224 +17,41 @@
 package v1alpha1
 
 import (
-	"fmt"
-
 	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
 	nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
-	"sigs.k8s.io/yaml"
-)
-
-const (
-	// errors
-	errKubeObjectNotInitialized = "KubeObject not initialized"
+	"github.com/nephio-project/nephio/krm-functions/lib/kubeobject"
 )
 
-var (
-	attachmentType      = []string{"spec", "attachmentType"}
-	cniType             = []string{"spec", "cniType"}
-	networkInstanceName = []string{"spec", "networkInstance", "name"}
-)
+type Interface struct {
```

Comment:
The value is you get explicit type checking for the object and we need this in several libraries. This is the reason to do this. otherwise people need to implement this in each function independently again and again.

Reply:
what I did not do so far is have the explicit SeptSpec and SetStatus with their specific type, so this would add even more type awareness.

Reply:
the way to look at this is a the specific KubeObj extensions for the InterfaceReq CRD